### PR TITLE
Replace GitHub action with bash oneliner

### PR DESCRIPTION
**Pull request by thomaseizinger**

### CODE REVIEW

**Review by luckysori — APPROVED** [review]
Nice, shall we test it in this PR?

**Comment by thomaseizinger** [comment]
> Nice, shall we test it in this PR?

Well it is clearly not working!

**Comment by klochowicz** [comment]
nice & clean!

**Review by klochowicz — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -26,13 +26,21 @@ jobs:
...
+        with:
+          fetch-depth: 0
+      - name: Get all commits on current master
+        run: git fetch origin master
+      - name: Log all commits we will analyze
+        run: git log --pretty=format:%s origin/master..HEAD
+      - name: Disallow fixup! commits
+        run: git log --pretty=format:%s origin/master..HEAD | grep -zv fixup!
+      - name: Disallow squash! commits
+        run: git log --pretty=format:%s origin/master..HEAD | grep -zv squash!
```
Comment (klochowicz) [inline .github/workflows/ci.yml:43]: it also opens the door to easily add more rules if we ever desire, e.g. length of the description etc.!

Comment (thomaseizinger) [reply .github/workflows/ci.yml:43]: Yep!

**Pull request merged**